File: SampleSqlLite/app/src/main/java/com/sqllite/samplesqllite/EditAnimal.java
package com.sqllite.samplesqllite;

import java.util.HashMap;

import android.app.Activity;
import android.content.Intent;
import android.os.Bundle;
import android.util.Log;
import android.view.View;
import android.widget.EditText;

public class EditAnimal extends Activity {
    EditText animalName;
    EditText animalTallness;
    DBController controller = new DBController(this);

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_edit_animal);
        animalName = findViewById(R.id.animalName);
        animalTallness = findViewById(R.id.animalTallness);
        Intent objIntent = getIntent();
        String animalId = objIntent.getStringExtra("animalId");
        Log.d("Reading: ", "Reading animals...");
        HashMap<String, String> animalList = controller.getAnimalInfo(animalId);
        Log.d("animalName", animalList.get("animalName"));
        Log.d("animalTallness", animalList.get("animalTallness"));
        if (animalList.size() != 0) {
            animalName.setText(animalList.get("animalName"));
            animalTallness.setText(animalList.get("animalTallness"));
        }
    }

    public void editAnimal(View view) {
        HashMap<String, String> queryValues = new HashMap<String, String>();
        animalName = findViewById(R.id.animalName);
        animalTallness = findViewById(R.id.animalTallness);

        Intent objIntent = getIntent();
        String animalId = objIntent.getStringExtra("animalId");
        queryValues.put("animalId", animalId);
        queryValues.put("animalName", animalName.getText().toString());
        queryValues.put("animalTallness", animalTallness.getText().toString());

        controller.updateAnimal(queryValues);
        this.callHomeActivity(view);

    }

    public void removeAnimal(View view) {
        Intent objIntent = getIntent();
        String animalId = objIntent.getStringExtra("animalId");
        controller.deleteAnimal(animalId);
        this.callHomeActivity(view);

    }

    public void callHomeActivity(View view) {
        Intent objIntent = new Intent(getApplicationContext(), MainActivity.class);
        startActivity(objIntent);
    }
}

File: kotlin/SampleSqlLite/app/src/main/java/com/dstrube/samplesqllite/EditAnimal.kt
package com.dstrube.samplesqllite

import android.content.Intent
import android.content.Intent.getIntent
import android.widget.EditText
import android.os.Bundle
import android.app.Activity
import android.util.Log
import android.view.View


class EditAnimal : Activity() {
    internal var animalName: EditText? = null
    internal var animalTallness: EditText? = null
    internal var controller = DBController(this)
    public override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_edit_animal)
        animalName = findViewById(R.id.animalName)
        animalTallness = findViewById(R.id.animalTallness)
        val objIntent = intent
        val animalId = objIntent.getStringExtra("animalId")
        Log.d("Reading: ", "Reading animals...")
        val animalList = animalId?.let { controller.getAnimalInfo(it) }
        if (animalList != null) {
            animalList["animalName"]?.let { Log.d("animalName", it) }
        }
        if (animalList != null) {
            animalList["animalTallness"]?.let { Log.d("animalTallness", it) }
        }

        if (animalList != null) {
            if (animalList.size > 0) {
                if (animalList != null) {
                    animalName?.setText(animalList["animalName"])
                }
                if (animalList != null) {
                    animalTallness?.setText(animalList["animalTallness"])
                }
            }
        }
    }

    fun editAnimal(view: View) {
        val queryValues = HashMap<String, String>()
        animalName = findViewById(R.id.animalName)
        animalTallness = findViewById(R.id.animalTallness)

        val objIntent = intent
        val animalId = objIntent.getStringExtra("animalId")
        if(animalId != null) {
            queryValues["animalId"] = animalId
        }
        queryValues["animalName"] = animalName?.text.toString()
        queryValues["animalTallness"] = animalTallness?.text.toString()

        controller.updateAnimal(queryValues)
        this.callHomeActivity(view)

    }

    fun removeAnimal(view: View) {
        val objIntent = intent
        val animalId = objIntent.getStringExtra("animalId")
        if (animalId != null) {
            controller.deleteAnimal(animalId)
        }
        this.callHomeActivity(view)

    }

    fun callHomeActivity(view: View) {
        val objIntent = Intent(applicationContext, MainActivity::class.java)
        startActivity(objIntent)
    }
}

